quic/s2n-quic-core/src/sync/spsc/primitive.rs

```diff
@@ -40,3 +40,14 @@ mod core {
 
 #[cfg(not(all(loom, test)))]
 pub use self::core::*;
+
+/// Indicates if the type is a zero-sized type
+///
+/// This can be used to optimize the code to avoid needless calculations.
+pub trait IsZst {
+    const IS_ZST: bool;
```

Comment:
Not going to block on this, but I would call this `IS_ZERO_SIZED`

Reply:
I was following the [stdlib](https://github.com/rust-lang/rust/blob/9545094994f1ab45cab5799d5b45980871a9e97b/library/alloc/src/collections/vec_deque/mod.rs#L274) convention here, since the design of the queue cursors is based on VecDeque.